File: 203RemoveLinkedListeElements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        newhead = pre = ListNode(0)
        pre.next = head
        cur = head

        while cur is not None:
        	if cur.val == val:
        		cur = cur.next
        		pre.next = cur
        	else:
        		pre = cur
        		cur = cur.next

        return newhead.next

File: 203RemoveLinkedListeElements.cpp
/*
Remove all elements from a linked list of integers that have value val.
Example:

Input:  1->2->6->3->4->5->6, val = 6
Output: 1->2->3->4->5

*/

/**
  * Definition for singly-linked list.
  * struct ListNode{
  * 	int val;
  *		ListNode *next;
  *		ListNode(int x) : val(x), next(NULL){}
  *};
  */

class Solution{
public:
	ListNode* removeElements(ListNode* head, int val){
		if(head == NULL) return head;
		ListNode *newhead = new ListNode(-1);
		newhead->next = head;
		ListNode *pre = newhead;
		ListNode *cur = head;
		while(cur != NULL){
			if(cur->val == val){
				cur = cur->next;
				pre->next = cur;
			}
			else{
				pre = cur;
				cur = cur->nxet;
			}
	
		}
		return newhead->next;
	}
}